python/dgl/distributed/graph_partition_book.py

```diff
@@ -658,9 +910,94 @@ def get_size(self):
         int
             data size
         """
-        if self._policy_str == EDGE_PART_POLICY:
-            return self._partition_book._num_edges()
-        elif self._policy_str == NODE_PART_POLICY:
-            return self._partition_book._num_nodes()
+        if EDGE_PART_POLICY in self._policy_str:
+            return self._partition_book._num_edges(self._policy_str[5:])
+        elif NODE_PART_POLICY in self._policy_str:
+            return self._partition_book._num_nodes(self._policy_str[5:])
         else:
             raise RuntimeError('Cannot support policy: %s ' % self._policy_str)
+
+class NodePartitionPolicy(PartitionPolicy):
+    '''Partition policy for nodes.
+    '''
+    def __init__(self, partition_book, ntype='_N'):
+        super(NodePartitionPolicy, self).__init__(NODE_PART_POLICY + ':' + ntype, partition_book)
+
+class EdgePartitionPolicy(PartitionPolicy):
+    '''Partition policy for edges.
+    '''
+    def __init__(self, partition_book, etype='_E'):
+        super(EdgePartitionPolicy, self).__init__(EDGE_PART_POLICY + ':' + etype, partition_book)
+
+class HeteroDataName(object):
+    ''' The data name in a heterogeneous graph.
+
+    A unique data name has three components:
+    * indicate it's node data or edge data.
+    * indicate the node/edge type.
+    * the name of the data.
```

Comment:
What does `data` precisely mean?  I assume it's features?

Also the term `data name` and the class name `HeteroDataName` is extremely vague, and may bring confusion with code like `HeteroDataName.get_name()` as you will be getting a name from a name.  Consider renaming it.

I would rename the class to something like `FeatureIdentifier` and put the docstring to something like:
```
A unique identifier for features on heterogeneous graphs, indicating

* Whether the feature is a node feature or edge feature.
* The name of the node/edge type the feature is on.
* The feature's name.
```

Reply:
I think "feature" isn't very general for the data stored on nodes and edges. We can store labels, masks, etc on nodes/edges. Do we refer to them as features as well?

Reply:
I notice that the doc string of DGLHeteroGraph uses "feature" a lot. Does it cause confusion?

Reply:
Let's stick to the name "feature" for now and change it once we hear someone complaining.